File: clients/ts/examples/live-smoke.js
import { OsolarLinkClient } from "../dist/index.js";

const apiKey = process.env.OSOLAR_API_KEY;
if (!apiKey) {
  console.error("OSOLAR_API_KEY is required");
  process.exit(1);
}

const client = new OsolarLinkClient({ apiKey });

try {
  const response = await client.listLinkedPlants();
  const count = Array.isArray(response.data) ? response.data.length : 0;
  console.log(
    JSON.stringify(
      {
        success: response.success,
        linkedPlantCount: count,
      },
      null,
      2,
    ),
  );
} catch (error) {
  console.error("Live smoke test failed:", error);
  process.exit(1);
}

File: clients/rust/examples/live_smoke.rs
use osolar_client::OsolarLinkClient;

fn main() {
    let api_key = std::env::var("OSOLAR_API_KEY").unwrap_or_else(|_| {
        eprintln!("OSOLAR_API_KEY is required");
        std::process::exit(1);
    });

    let client = OsolarLinkClient::new(api_key);

    match client.list_linked_plants() {
        Ok(response) => {
            let count = response.data.as_ref().map(std::vec::Vec::len).unwrap_or(0);
            println!(
                "{{\"success\":{},\"linkedPlantCount\":{}}}",
                response.success, count
            );
        }
        Err(err) => {
            eprintln!("Live smoke test failed: {err}");
            std::process::exit(1);
        }
    }
}
